rpcs3/rpcs3qt/cheat_manager.cpp

```diff
@@ -245,12 +245,12 @@ bool cheat_engine::resolve_script(u32& final_offset, const u32 offset, const std
 
 	while (index < red_script.size())
 	{
-		if (red_script[index] >= '0' && red_script[index] <= '9')
+		if (std::isdigit(red_script[index]))
 		{
 			std::string num_string;
 			for (; index < red_script.size(); index++)
 			{
-				if (red_script[index] < '0' || red_script[index] > '9')
+				if (!std::isdigit(red_script[index]))
```

Comment:
```suggestion
				if (!std::isdigit(static_cast<u8>(red_script[index])))
```
This fixes the UB mentioned earlier.